File: 프로그래머스/lv2/42578. 의상/의상.js
function solution(clothes) {
    var answer = 1;
    const clothMap = new Map();
    clothes.forEach((cloth) => {
        if(clothMap.has(cloth[1])) {
            clothMap.set(cloth[1], clothMap.get(cloth[1]) + 1)
        } else {
            clothMap.set(cloth[1], 1)
        }
    })
    for(const [key, value] of clothMap.entries()){
        answer *= (value + 1)
    }
    return answer - 1;
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1;
    closet = {}
    for cloth in clothes:
        if cloth[1] not in closet:
            closet.setdefault(cloth[1], 1)
        else:
            closet[cloth[1]] += 1
    for key, value in closet.items():
        answer *= (value + 1)
    return answer - 1